examples/gno.land/r/demo/gnolotto_factory/gnolotto_factory.gno

```diff
@@ -0,0 +1,229 @@
+package gnolotto_factory
+
+import (
+	"bytes"
+	"std"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/gnolotto"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/demo/seqid"
+)
+
+var lotteries *avl.Tree
+var lotteriesID seqid.ID
+
+// Replace this address with your address
+var admin std.Address = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+
+const TICKET_PRICE = 10
+
+// Initializes the lottery AVL tree
+func init() {
+	lotteries = avl.NewTree()
+}
+
+// Creates a new lottery, only callable by admin.
+func CreateLottery(drawTime int64, prizePool int64) (string, string) {
+	sentCoins := std.GetOrigSend()
+	amount := sentCoins.AmountOf("ugnot")
+
+	if prizePool != amount {
+		panic("Prize pool must match the transaction value")
+	}
+
+	if drawTime < time.Now().Unix() {
+		panic("Invalid draw time")
+	}
+
+	if std.GetOrigCaller() != admin {
+		panic("Only the admin can create a lottery")
+	}
+
+	lotteryID := lotteriesID.Next()
+	lottery := gnolotto.NewLottery(time.Unix(drawTime, 0), prizePool)
+
+	lotteries.Set(lotteryID.Binary(), lottery)
+	return lotteryID.String(), "Lottery created successfully"
+}
+
+// Buy ticket for a specific lottery.
+func BuyTicket(lotteryIDStr string, numbersStr string) (int, string) {
+	sentCoins := std.GetOrigSend()
+	amount := sentCoins.AmountOf("ugnot")
+
+	lotteryID, _ := seqid.FromString(lotteryIDStr)
+	id := lotteryID.Binary()
+
+	lotteryRaw, exists := lotteries.Get(id)
+	if !exists {
+		panic("Lottery not found")
+	}
+
+	// Convert string to slice of integers.
+	numbers, err := gnolotto.StringToIntSlice(numbersStr)
+	if err != nil {
+		panic("Invalid number: " + err.Error()) // Gestion de l'erreur si la conversion échoue.
+	}
+
+	//Verify if the amount sent is equal to the ticket price.
+	if amount != TICKET_PRICE {
+		panic("Ticket price must be 10 UGNOT")
+	}
+
+	// Verify if the numbers are unique.
+	uniqueNumbers := make(map[int]bool)
+	for _, num := range numbers {
+		if uniqueNumbers[num] {
+			panic("Numbers must be unique")
+		}
+		uniqueNumbers[num] = true
+	}
+
+	l, _ := lotteryRaw.(*gnolotto.Lottery)
+
+	if time.Now().Unix() > l.DrawTime.Unix() {
+		panic("This lottery has already ended")
+	}
+
+	if len(numbers) != gnolotto.MaxLottoNumbers {
+		panic("You must select exactly 5 numbers")
+	}
+
+	for _, num := range numbers {
+		if num > 15 || num < 1 {
+			panic("Invalid number, select number range from 1 to 15")
+		}
+	}
+
+	caller := std.GetOrigCaller()
+	l.AddTicket(numbers, caller)
+
+	return 1, "Ticket purchased successfully"
```

Comment:
Just like a previous comment, you don't have to return all of these, especially since you've switched to panicking - now, if the function does not panic, it means that it was executed correctly and without errors, meaning you don't really need this `1`. Just return the string informing the user that the ticket went through.